### ISSUE TITLE
Std is bitflags in error message

### ISSUE THREAD

**Issue by parkovski** [rust-lang/rust#44864]
For a program like this:
```
#[macro_use]
extern crate bitflags;

bitflags! {
    #[derive(Default)]
    pub struct Flags: u32 {
        const FLAG = 0x1;
        const MORE_FLAGS = 0x2;
    }
}

fn its_a_me_bitflags(flags: Flags) -> Option<Flags> {
    Some(flags)
}

fn pointless_fn(flags: Flags) -> Result<Flags, ()> {
    its_a_me_bitflags(flags)?;
    Ok(flags)
}

fn main() {
    match pointless_fn(Flags::FLAG) {
        Ok(f) => println!("{}", f),
        Err(()) => println!("What does an error even mean here?"),
    }
}
```
I'm getting errors that look like this with the last few nightlies:
```
   Compiling r v0.1.0 (file:///C:/Users/parke/Documents/exp/r)
error[E0277]: the `?` operator can only be applied to values that implement `bitflags::<unnamed>::ops::Try`
  --> src\main.rs:17:5
   |
17 |     its_a_me_bitflags(flags)?;
   |     -------------------------
   |     |
   |     the `?` operator cannot be applied to type `bitflags::<unnamed>::option::Option<Flags>`
   |     in this macro invocation
   |
   = help: the trait `bitflags::<unnamed>::ops::Try` is not implemented for `bitflags::<unnamed>::option::Option<Flags>`
   = note: required by `bitflags::<unnamed>::ops::Try::into_result`
```
Running on Windows 10 x64, msvc build, happens with the latest nightly as well as the one I had before from a couple days ago or so.

This doesn't happen on the Linux build for me, only Windows.

**Comment by ChrisDenton** [comment]
This seems to have been fixed? Using edition 2015 and `bitflags = "= 1.0.0"` the error output on Windows msvc does not have `bitflags::<unnamed>::` etc:

```plain
warning: use of deprecated macro `try`: use the `?` operator instead
  --> src/main.rs:4:1
   |
4  | / bitflags! {
5  | |     #[derive(Default)]
6  | |     pub struct Flags: u32 {
7  | |         const FLAG = 0x1;
8  | |         const MORE_FLAGS = 0x2;
9  | |     }
10 | | }
   | |_^
   |
   = note: `#[warn(deprecated)]` on by default
   = note: this warning originates in the macro `__impl_bitflags` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of deprecated macro `try`: use the `?` operator instead
  --> src/main.rs:4:1
   |
4  | / bitflags! {
5  | |     #[derive(Default)]
6  | |     pub struct Flags: u32 {
7  | |         const FLAG = 0x1;
8  | |         const MORE_FLAGS = 0x2;
9  | |     }
10 | | }
   | |_^
   |
   = note: this warning originates in the macro `__impl_bitflags` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the `?` operator can only be used on `Result`s, not `Option`s, in a function that returns `Result`
   --> src/main.rs:17:29
    |
16  | / fn pointless_fn(flags: Flags) -> Result<Flags, ()> {
17  | |     its_a_me_bitflags(flags)?;
    | |                             ^ use `.ok_or(...)?` to provide an error compatible with `Result<Flags, ()>`
18  | |     Ok(flags)
19  | | }
    | |_- this function returns a `Result`
    |
    = help: the trait `FromResidual<Option<Infallible>>` is not implemented for `Result<Flags, ()>`
note: required by `from_residual`

error[E0277]: `Flags` doesn't implement `std::fmt::Display`
  --> src/main.rs:23:33
   |
23 |         Ok(f) => println!("{}", f),
   |                                 ^ `Flags` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `Flags`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: this error originates in the macro `$crate::format_args_nl` (in Nightly builds, run with -Z macro-backtrace for more info)
```

For reference, compiling for Linux gives:

```
warning: use of deprecated macro `try`: use the `?` operator instead
  --> src/main.rs:4:1
   |
4  | / bitflags! {
5  | |     #[derive(Default)]
6  | |     pub struct Flags: u32 {
7  | |         const FLAG = 0x1;
8  | |         const MORE_FLAGS = 0x2;
9  | |     }
10 | | }
   | |_^
   |
   = note: `#[warn(deprecated)]` on by default
   = note: this warning originates in the macro `__impl_bitflags` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of deprecated macro `try`: use the `?` operator instead
  --> src/main.rs:4:1
   |
4  | / bitflags! {
5  | |     #[derive(Default)]
6  | |     pub struct Flags: u32 {
7  | |         const FLAG = 0x1;
8  | |         const MORE_FLAGS = 0x2;
9  | |     }
10 | | }
   | |_^
   |
   = note: this warning originates in the macro `__impl_bitflags` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the `?` operator can only be used on `Result`s, not `Option`s, in a function that returns `Result`
   --> src/main.rs:17:29
    |
16  | / fn pointless_fn(flags: Flags) -> Result<Flags, ()> {
17  | |     its_a_me_bitflags(flags)?;
    | |                             ^ use `.ok_or(...)?` to provide an error compatible with `Result<Flags, ()>`
18  | |     Ok(flags)
19  | | }
    | |_- this function returns a `Result`
    |
    = help: the trait `FromResidual<Option<Infallible>>` is not implemented for `Result<Flags, ()>`
note: required by `from_residual`

error[E0277]: `Flags` doesn't implement `std::fmt::Display`
  --> src/main.rs:23:33
   |
23 |         Ok(f) => println!("{}", f),
   |                                 ^ `Flags` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `Flags`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: this error originates in the macro `$crate::format_args_nl` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by ChrisDenton** [comment]
This seems to have been fixed? Using edition 2015 and `bitflags = "= 1.0.0"` the error output on Windows msvc does not have `bitflags::<unnamed>::` etc:

```plain
warning: use of deprecated macro `try`: use the `?` operator instead
  --> src/main.rs:4:1
   |
4  | / bitflags! {
5  | |     #[derive(Default)]
6  | |     pub struct Flags: u32 {
7  | |         const FLAG = 0x1;
8  | |         const MORE_FLAGS = 0x2;
9  | |     }
10 | | }
   | |_^
   |
   = note: `#[warn(deprecated)]` on by default
   = note: this warning originates in the macro `__impl_bitflags` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of deprecated macro `try`: use the `?` operator instead
  --> src/main.rs:4:1
   |
4  | / bitflags! {
5  | |     #[derive(Default)]
6  | |     pub struct Flags: u32 {
7  | |         const FLAG = 0x1;
8  | |         const MORE_FLAGS = 0x2;
9  | |     }
10 | | }
   | |_^
   |
   = note: this warning originates in the macro `__impl_bitflags` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the `?` operator can only be used on `Result`s, not `Option`s, in a function that returns `Result`
   --> src/main.rs:17:29
    |
16  | / fn pointless_fn(flags: Flags) -> Result<Flags, ()> {
17  | |     its_a_me_bitflags(flags)?;
    | |                             ^ use `.ok_or(...)?` to provide an error compatible with `Result<Flags, ()>`
18  | |     Ok(flags)
19  | | }
    | |_- this function returns a `Result`
    |
    = help: the trait `FromResidual<Option<Infallible>>` is not implemented for `Result<Flags, ()>`
note: required by `from_residual`

error[E0277]: `Flags` doesn't implement `std::fmt::Display`
  --> src/main.rs:23:33
   |
23 |         Ok(f) => println!("{}", f),
   |                                 ^ `Flags` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `Flags`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: this error originates in the macro `$crate::format_args_nl` (in Nightly builds, run with -Z macro-backtrace for more info)
```

For reference, compiling for Linux gives:

<details>

````
warning: use of deprecated macro `try`: use the `?` operator instead
  --> src/main.rs:4:1
   |
4  | / bitflags! {
5  | |     #[derive(Default)]
6  | |     pub struct Flags: u32 {
7  | |         const FLAG = 0x1;
8  | |         const MORE_FLAGS = 0x2;
9  | |     }
10 | | }
   | |_^
   |
   = note: `#[warn(deprecated)]` on by default
   = note: this warning originates in the macro `__impl_bitflags` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of deprecated macro `try`: use the `?` operator instead
  --> src/main.rs:4:1
   |
4  | / bitflags! {
5  | |     #[derive(Default)]
6  | |     pub struct Flags: u32 {
7  | |         const FLAG = 0x1;
8  | |         const MORE_FLAGS = 0x2;
9  | |     }
10 | | }
   | |_^
   |
   = note: this warning originates in the macro `__impl_bitflags` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the `?` operator can only be used on `Result`s, not `Option`s, in a function that returns `Result`
   --> src/main.rs:17:29
    |
16  | / fn pointless_fn(flags: Flags) -> Result<Flags, ()> {
17  | |     its_a_me_bitflags(flags)?;
    | |                             ^ use `.ok_or(...)?` to provide an error compatible with `Result<Flags, ()>`
18  | |     Ok(flags)
19  | | }
    | |_- this function returns a `Result`
    |
    = help: the trait `FromResidual<Option<Infallible>>` is not implemented for `Result<Flags, ()>`
note: required by `from_residual`

error[E0277]: `Flags` doesn't implement `std::fmt::Display`
  --> src/main.rs:23:33
   |
23 |         Ok(f) => println!("{}", f),
   |                                 ^ `Flags` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `Flags`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: this error originates in the macro `$crate::format_args_nl` (in Nightly builds, run with -Z macro-backtrace for more info)
````

</details>
